### PR TITLE
feat(bottom nav): Add bottom navigation on mobile

### PR DESCRIPTION
**Pull request by 0xCorgi**
[Current preview](https://corgi-sparkle.surge.sh/?path=/story/widgets-menu--connected)

New components:
- BottomNav
- BottomNavItem
- IconComponent
- AnimatedIconComponent

The BottomNav has 2 specials behavior:
- tap and hold to display menu (only on a touch device)
- animated icon on menu change

Icons from the menu, along with new ones, have been added to the main icons folder.

image

### CODE REVIEW

File: packages/pancake-uikit/src/components/Svg/Icons/Earn.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import * as React from "react";
+import Svg from "../Svg";
+import { SvgProps } from "../types";
+
+const Icon: React.FC<SvgProps> = (props) => {
+  return (
+    <Svg viewBox="0 0 22 20" {...props}>
```
Comment (hachiojidev) [inline packages/pancake-uikit/src/components/Svg/Icons/Earn.tsx:7]: This should be 24x24.

Comment (hachiojidev) [reply packages/pancake-uikit/src/components/Svg/Icons/Earn.tsx:7]: I added this icon just now. Go ahead and rebase.

**Review by hachiojidev — CHANGES_REQUESTED** [review]
Just rebase to make sure icons are still good.

**Comment by hachiojidev** [comment]
@0xCorgi Please make sure to rebase. I see a lot of duplicate commits.

**Comment by 0xCorgi** [comment]
> @0xCorgi Please make sure to rebase. I see a lot of duplicate commits.

This PR will be a nightmare to rebase. I think it would be best to squash the commit on this one.
I already merge your changes to make sure the icons are still good.
I'll rebase the final PR at the end anyways.

**Pull request merged**